sentry-async-profiler/src/main/java/io/sentry/asyncprofiler/convert/JfrAsyncProfilerToSentryProfileConverter.java

```diff
@@ -0,0 +1,169 @@
+                        metadata.priority = 0;
+                        return metadata;
+                      });
+                }
+              }
+
+              if (sentryProfile.samples == null) {
+                sentryProfile.samples = new ArrayList<>();
+              }
+
+              if (sentryProfile.frames == null) {
+                sentryProfile.frames = new ArrayList<>();
+              }
+
+              List<Integer> stack = new ArrayList<>();
+              int currentStack = stacks.size();
+              int currentFrame = sentryProfile.frames != null ? sentryProfile.frames.size() : 0;
+              for (int i = 0; i < methods.length; i++) {
+                //          for (int i = methods.length; --i >= 0; ) {
+                SentryStackFrame frame = new SentryStackFrame();
+                StackTraceElement element =
+                    getStackTraceElement(methods[i], types[i], locations[i]);
+                if (element.isNativeMethod()) {
+                  continue;
+                }
+
+                final String classNameWithLambdas = element.getClassName().replace("/", ".");
+                frame.setFunction(element.getMethodName());
+
+                int firstDollar = classNameWithLambdas.indexOf('$');
+                String sanitizedClassName = classNameWithLambdas;
+                if (firstDollar != -1) {
+                  sanitizedClassName = classNameWithLambdas.substring(0, firstDollar);
+                }
+
+                int lastDot = sanitizedClassName.lastIndexOf('.');
+                if (lastDot > 0) {
+                  frame.setModule(sanitizedClassName);
+                } else if (!classNameWithLambdas.startsWith("[")) {
+                  frame.setModule("");
+                }
+
+                if (element.isNativeMethod() || classNameWithLambdas.isEmpty()) {
+                  frame.setInApp(false);
+                } else {
+                  frame.setInApp(
+                      new SentryStackTraceFactory(Sentry.getGlobalScope().getOptions())
+                          .isInApp(sanitizedClassName));
+                }
+
+                frame.setLineno((element.getLineNumber() != 0) ? element.getLineNumber() : null);
+                frame.setFilename(classNameWithLambdas);
+
+                if (sentryProfile.frames != null) {
+                  sentryProfile.frames.add(frame);
+                }
+                stack.add(currentFrame);
+                currentFrame++;
+              }
+
+              long divisor = jfr.ticksPerSec / 1000_000_000L;
+              long myTimeStamp =
+                  jfr.chunkStartNanos + ((event.time - jfr.chunkStartTicks) / divisor);
+
+              JfrSample sample = new JfrSample();
+              Instant instant = Instant.ofEpochSecond(0, myTimeStamp);
+              double timestampDouble =
+                  instant.getEpochSecond() + instant.getNano() / 1_000_000_000.0;
+
+              sample.timestamp = timestampDouble;
```

Comment:
Revisiting this in follow-up PR, I think the way the timestamp is calculated is not correct